### PR TITLE
Committee query improvements

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

This was initially meant to be a test to verify the behavior reported in this issue: https://github.com/IntersectMBO/cardano-ledger/issues/4001
However  I realized that we could improve the query by returning  results independent of whether there exists a current committee, since it's possible that we have relevant information even when there is no currently elected committee (for example: information about the next epoch committee, or about pre-authorized members). 
It's possible that returning better results helps with clarifying the behavior reported in the bug, for which I added a test in the last commit. 

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs [hunk 1]
```diff
@@ -184,87 +177,96 @@ queryCommitteeMembersState ::
...
   AccountState
 queryAccountState = view $ nesEsL . esAccountStateL
+
+getNextEpochCommitteeMembers ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  Map (Credential 'ColdCommitteeRole (EraCrypto era)) EpochNo
+getNextEpochCommitteeMembers nes =
+  let ratifyState = snd . finishDRepPulser $ queryGovState nes ^. drepPulsingStateGovStateL
+      committee = strictMaybeToMaybe $ ratifyState ^. rsEnactStateL . ensCommitteeL
```
Comment (lehins) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs]: Correct me if I am wrong, but I don't think this conversion is necessary
```suggestion
      committee = ratifyState ^. rsEnactStateL . ensCommitteeL
```

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs [hunk 2]
```diff
@@ -184,87 +177,96 @@ queryCommitteeMembersState ::
...
-        csCommitteeCreds $
-          nes ^. nesEpochStateL . esLStateL . lsCertStateL . certVStateL . vsCommitteeStateL
-
-      withFilteredColdCreds s
-        | Set.null coldCredsFilter = s
-        | otherwise = s `Set.intersection` coldCredsFilter
+  CommitteeMembersState (EraCrypto era)
+queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes =
+  let
+    committee = strictMaybeToMaybe $ queryGovState nes ^. committeeGovStateL
```
Comment (lehins) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs]: Probably unnecessary 
```suggestion
    committee = queryGovState nes ^. committeeGovStateL
```

**Review by lehins — APPROVED** [review]
Looks great. Thank you!

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs (hunk 2 above)
Comment (teodanciu) [reply libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs]: Done, thanks!

**Pull request merged**